beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -680,12 +684,37 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
 
     /// Returns the block canonical root of the current canonical chain at a given slot.
     ///
-    /// Returns None if a block doesn't exist at the slot.
+    /// Returns `None` if the given slot doesn't exist in the chain.
     pub fn root_at_slot(&self, target_slot: Slot) -> Result<Option<Hash256>, Error> {
-        process_results(self.rev_iter_block_roots()?, |mut iter| {
-            iter.find(|(_, slot)| *slot == target_slot)
-                .map(|(root, _)| root)
-        })
+        let head = self.head()?;
```

Comment:
> Am I right that the purpose of this change is to use BlockRootsIterator::owned?

Think you mean "not use" but yes you're right. 

> If so, we don't get any gains from using BlockRootsIterator::new here since self.head() returns  a clone of the head. We own the beacon state, so it's equivalent to use BlockRootsIterator::owned.

Ah dang, didn't notice that, will revert it.

> Without this iter::once we will never visit the block at the head of the chain.

I did notice the `iter::once` but didn't understand the reason, makes sense. 👍